backfort of fix for default value of btl_openib_memalign_threshold, see https://github.com/open-mpi/ompi/issues/638
author: Kenneth Hoste
diff -ru openmpi-1.7.3.orig/ompi/mca/btl/openib/btl_openib_component.c openmpi-1.7.3/ompi/mca/btl/openib/btl_openib_component.c
--- openmpi-1.7.3.orig/ompi/mca/btl/openib/btl_openib_component.c	2013-09-26 20:23:06.000000000 +0200
+++ openmpi-1.7.3/ompi/mca/btl/openib/btl_openib_component.c	2015-07-15 08:42:50.000000000 +0200
@@ -2477,7 +2477,9 @@
        we're most likely going to be using this component). The hook is to be set up
        as early as possible in this function since we want most of the allocated resources
        be aligned.*/
-    if (mca_btl_openib_component.use_memalign > 0) {
+    if (mca_btl_openib_component.use_memalign > 0 &&
+        (opal_mem_hooks_support_level() &
+            (OPAL_MEMORY_FREE_SUPPORT | OPAL_MEMORY_CHUNK_SUPPORT)) != 0) {
         mca_btl_openib_component.previous_malloc_hook = __malloc_hook;
         __malloc_hook = btl_openib_malloc_hook; 
         malloc_hook_set = true;
diff -ru openmpi-1.7.3.orig/ompi/mca/btl/openib/btl_openib_mca.c openmpi-1.7.3/ompi/mca/btl/openib/btl_openib_mca.c
--- openmpi-1.7.3.orig/ompi/mca/btl/openib/btl_openib_mca.c	2013-09-26 20:23:06.000000000 +0200
+++ openmpi-1.7.3/ompi/mca/btl/openib/btl_openib_mca.c	2015-07-15 08:42:05.000000000 +0200
@@ -631,7 +631,7 @@
                   32, &mca_btl_openib_component.use_memalign,
                   REGINT_GE_ZERO));
 
-    mca_btl_openib_component.memalign_threshold = mca_btl_openib_component.eager_limit;
+    mca_btl_openib_component.memalign_threshold = mca_btl_openib_module.super.btl_eager_limit;
     tmp = mca_base_component_var_register(&mca_btl_openib_component.super.btl_version,
                                           "memalign_threshold",
                                           "Allocating memory more than btl_openib_memalign_threshhold"
--- openmpi-1.7.3.orig/opal/etc/openmpi-mca-params.conf.orig 2015-06-05 16:21:25.535727000 +0200
+++ openmpi-1.7.3/opal/etc/openmpi-mca-params.conf  2015-06-08 10:36:05.387902000 +0200
@@ -56,3 +56,6 @@
 
 # See "ompi_info --param all all" for a full listing of Open MPI MCA
 # parameters available and their default values.
+
+# Fix bug in the initialization of the value when using the infiniband module. (ref: http://www.open-mpi.org/community/lists/users/2015/05/26913.php)
+btl_openib_memalign_threshold=12288
